app/controllers/ClaimPeriodQuestionController.scala

```diff
@@ -0,0 +1,85 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import java.time.LocalDate
+
+import config.FrontendAppConfig
+import controllers.actions._
+import forms.ClaimPeriodQuestionFormProvider
+import handlers.{ClaimPeriodQuestionRequestHandler, ErrorHandler}
+import javax.inject.Inject
+import models.ClaimPeriodQuestion
+import models.requests.DataRequest
+import navigation.Navigator
+import pages.{ClaimPeriodEndPage, ClaimPeriodQuestionPage, ClaimPeriodStartPage}
+import play.api.data.Form
+import play.api.i18n.MessagesApi
+import play.api.mvc._
+import repositories.SessionRepository
+import views.html.ClaimPeriodQuestionView
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class ClaimPeriodQuestionController @Inject()(
+  override val messagesApi: MessagesApi,
+  sessionRepository: SessionRepository,
+  val navigator: Navigator,
+  identify: IdentifierAction,
+  config: FrontendAppConfig,
+  getData: DataRetrievalAction,
+  requireData: DataRequiredAction,
+  formProvider: ClaimPeriodQuestionFormProvider,
+  val controllerComponents: MessagesControllerComponents,
+  view: ClaimPeriodQuestionView,
+)(implicit ec: ExecutionContext, errorHandler: ErrorHandler)
+    extends BaseController with ClaimPeriodQuestionRequestHandler {
+
+  val form = formProvider()
+
+  def onPageLoad(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+    getRequiredAnswers(ClaimPeriodStartPage, ClaimPeriodEndPage) { (claimStart, claimEnd) =>
+      val filledForm: Form[ClaimPeriodQuestion] =
+        request.userAnswers.get(ClaimPeriodQuestionPage).fold(form)(form.fill)
+
+      onLoadResult(claimStart, claimEnd, filledForm)
+    }
+  }
+
+  def onSubmit(): Action[AnyContent] = (identify andThen getData andThen requireData).async { implicit request =>
+    getRequiredAnswers(ClaimPeriodStartPage, ClaimPeriodEndPage) { (claimStart, claimEnd) =>
+      form
+        .bindFromRequest()
+        .fold(
+          formWithErrors => Future.successful(BadRequest(view(formWithErrors, claimStart, claimEnd))),
+          value => onSubmitResult(request, value)
+        )
+    }
+  }
+
+  private def onSubmitResult(request: DataRequest[AnyContent], value: ClaimPeriodQuestion): Future[Result] =
+    persistAnswer(request.userAnswers, value, sessionRepository.set).map(updatedAnswers =>
+      Redirect(navigator.nextPage(ClaimPeriodQuestionPage, updatedAnswers)))
+
+  private def onLoadResult(claimStart: LocalDate, claimEnd: LocalDate, filledForm: Form[ClaimPeriodQuestion])(
+    implicit request: Request[_]): Future[Result] =
+    if (config.fastTrackJourneyEnabled)
+      Future.successful(Ok(view(filledForm, claimStart, claimEnd)))
+    else
+      Future.successful(Redirect(routes.ClaimPeriodStartController.onPageLoad()))
```

Comment:
in this case it should be 404 as users are not supposed to hit the page at all

Reply:
this comment and the next two are all related - 
This controller is called once the button is clicked on confirmation page so we cannot do 404 but we simply go to claim-date as current behaviour - I think is fine as it is

Reply:
I agree this isn't quite consistent, but I think this is a slightly different case and is fine as-is.